agent/api/containerstatus.go

```diff
@@ -22,6 +22,13 @@ const (
 	ContainerCreated
 	// ContainerRunning represents a container that has started
 	ContainerRunning
+	// ContainerResourcesProvisioned represents a container that has completed provisioning all of its
+	// resources. Non-internal containers (containers present in the task definition) transition to
```

Comment:
That's fair. I had an offline chat with @samuelkarp  yesterday and decided to go with a modified approach for this. We'll be adding the ability for containers to define which is their steady state transition. Customer defined containers will have it set to `Running`, where as `internal` containers will have it set to different states based on why the internal container is being launched. The `pause` container will have the `ResourcesProvisioned` state as its `steady` state, where as the `volumes` container can have it set to `Created`. This also means that we need to change the logic in [Task Manager](https://github.com/aws/amazon-ecs-agent/blob/dev/agent/engine/task_manager.go#L378) code to be aware of these custom transitions as well.